.github/workflows/build-push-docker.yml

```
@@ -0,0 +1,65 @@
+name: Build and Push Subgraph Deployer Docker Image
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+'
+  workflow_dispatch:
+ 
+jobs:
+  # TODO: See how to ensure that the default CI workflow has run before this workflow
+  build-and-push:
+    if: |
+      (github.event_name == 'workflow_dispatch' && (github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/heads/feature/'))) ||
+      (github.event_name == 'push' && (startsWith(github.ref, 'refs/heads/release/') || startsWith(github.ref, 'refs/heads/main')))
```

```suggestion
      (
        github.event_name == 'workflow_dispatch' && (
          github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/heads/feature/')
        )
      ) ||
      (
        github.event_name == 'push' && (
          startsWith(github.ref, 'refs/heads/release/') || startsWith(github.ref, 'refs/heads/main')
        )
      )
```
Trying to make it more readable.